### ISSUE TITLE
get --format shell should preserve quotes

### ISSUE THREAD

**Issue by benjaminbauer** [dotenvx/dotenvx#392]
# Pre
given a `.env` file with
```
PASSWORD="f'bar"
```
# Action
`dotenvx get --format shell`

# Result
```
PASSWORD=f'bar
```

# Preferred result
(Ideally also for encrypted values)
```
PASSWORD="f'bar"
```

# Use Case
Passwords can and should contain all kinds of special characters. I want to use the output of `dotenvx get` in shell scripts. Removing the quotes requires me to "re-escape" values.

**Comment by motdotla** [comment]
this *should* work as-is in your shell script though:

```
$ echo "HELLO=\"f'bar\"" > .env
$ echo "console.log('Hello ' + process.env.HELLO)" > index.js
$ env $(dotenvx get --format shell) node index.js
Hello f'bar
```

**Comment by benjaminbauer** [comment]
True., but does not solve my use case currently.

I want to populate the shell env in a devcontainer with the vars from dotenvx. This removes the need to prepend every command with dotenvx run or similar.

Additional challenge: I have variables with a value, that contains a variable. 

So I tried `set -a && source <(dotenvx decrypt --stdout) && set +a`, but that does not work with variable expansion.

Another approach is to loop over the output of `dotenvx get --format shell` and call `export` on each. That gets involved quickly, because I would have to reliably split the key/value pairs (values can contain spaces or `=`). A great option would imo be, to allow to override the separator/delimiter for `dotenvx get --format shell`. e.g. `dotenvx get --format shell --separator='\n'`

`eval` does not work reliably, because the of the problem in my original post w/ unescaped special characters

**Comment by motdotla** [comment]
> I want to populate the shell env in a devcontainer with the vars from dotenvx. This removes the need to prepend every command with dotenvx run or similar.

can you elaborate? why can't you take the scenario i mention above and apply it to loading env to the devcontainer? what does your command look like?

**Comment by benjaminbauer** [comment]
Not really a command. In VS Code one can opt to reopen a project in a devcontainer. VS Code then builds/starts a docker container an connects a VS Code client to a VS Code server running in it. I do several things in the setup after a connection to the spun up container is established, e.g. install dotenvx. I also put a script in `/etc/profile.d/` that should call dotenvx and export all the env vars, so they can be used in any interactive shell, that a developer may open and for all extensions loaded in VS Code.

So long story short: controlling the startup of the container might be possible, but not the canonical way of doing devcontainers. So I want to use dotenvx in a script that is executed as part of the shell initialisation.

**Comment by motdotla** [comment]
that's cool.

ok, let me dive deeper into this. it's a tricky one because of differences in `env` vs `eval` and also supporting different shells.

you're idea of a separator is a fair one. i'll consider that.

until then, maybe just use `dotenvx get` (json) and then use `jq` to parse and set what you want?

**Comment by motdotla** [comment]
as i research, env is quite limited but also quite safe. eval is quite powerful at handling this but also quite dangerous. currently torn on which to service as priority for shell format. 

somewhat related: https://github.com/docker/cli/issues/3630

**Comment by motdotla** [comment]
opted to add a `--format eval` flag as it:

1. indicates to the user it could be more dangerous
2. works with `eval`'s more powerful parsing - allowing for encoding all types of odd characters

https://github.com/dotenvx/dotenvx/pull/393

going out now as `v1.17.0`

i think this will work for you but please comment if not.

**Comment by benjaminbauer** [comment]
Thanks you so much for your thorough consideration and instant implementation!

I think it can save some headaches to add some examples to the help/documentation for the new `format`?

My current best approach to putting dotenvx (w/ variable expansion in the variables values) into the env in a script is:
```bash
eval export $(dotenvx get --format eval)
```

This also works but has the gotacha/feature of **not overriding existing values**
```bash
set -a && eval $(dotenvx get --format eval) && set +a
```

**Comment by motdotla** [comment]
https://dotenvx.com/docs/advanced/get-eval and also in the README.

(you shouldn't need export in your first example - unless you want it the variables available to child processes (which i guess in this vscode case might be true))

you're welcome! thank you for using dotenvx!

**Comment by benjaminbauer** [comment]
> (you shouldn't need export in your first example - unless you want it the variables available to child processes (which i guess in this vscode case might be true))

That was the starting point of my quest here 😄
